cmd/subctl/diagnose.go

```diff
@@ -195,6 +194,9 @@ func addDiagnoseFirewallSubCommands() {
 	diagnoseFirewallNatDiscoveryRestConfigProducer.SetupFlags(diagnoseFirewallNatDiscovery.Flags())
 	addDiagnoseFWConfigFlags(diagnoseFirewallNatDiscovery)
 
+	diagnose.AddFirewallImageOverrideFlag(diagnoseFirewallVxLANCmd.Flags())
+	diagnose.AddFirewallImageOverrideFlag(diagnoseFirewallTunnelCmd.Flags())
+	diagnose.AddFirewallImageOverrideFlag(diagnoseFirewallNatDiscovery.Flags())
```

Comment:
Does this work with all three variants? IME sharing a flag variable in this way only works with one of the commands, that’s why there’s a `kube-proxy` variant and a `firewall` variant. I think the fix should be to change the original

    diagnose.AddFirewallImageOverrideFlag(diagnoseFirewallCmd.Flags())

line to

    diagnose.AddFirewallImageOverrideFlag(diagnoseFirewallCmd.PersistentFlags())